### PR TITLE
[pulsar-client-tools] Ability to send a message with null body

### PR DESCRIPTION
**Pull request by Shawyeok**
### Motivation

Now it's not possible use `pulsar-client produce` send a message with `null` body, but it's needed in some scenes, e.g. delete topic policy manually or test topic compaction.

### Modifications

Add an explicit option `--null` to `pulsar-client produce` subcommand.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change added tests and can be verified as follows:

  - *Added unit test PulsarClientToolTest#testNullBody*

### Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

- [x] `doc-added`
(Docs have been already added)

### CODE REVIEW

**Comment by poorbarcode** [comment]
> but it's needed in some scenes

Can you descript on that

**Review by Jason918 — COMMENTED** [review]
LGTM. Please make sure `CmdConsume` can handle null values.

**Comment by momo-jun** [comment]
Hi @Shawyeok, will you update the docs in a follow-up PR?

**Comment by Shawyeok** [comment]
> Hi @Shawyeok, will you update the docs in a follow-up PR?

I'm not sure this doc [reference-cli-tools](https://pulsar.apache.org/docs/next/reference-cli-tools#pulsar-client) is auto generated or not.

**Comment by Shawyeok** [comment]
> LGTM. Please make sure `CmdConsume` can handle null values.

Confirmed.

**Comment by momo-jun** [comment]
> > Hi @Shawyeok, will you update the docs in a follow-up PR?
> 
> @momo-jun I'm not sure this doc [reference-cli-tools](https://pulsar.apache.org/docs/next/reference-cli-tools#pulsar-client) is auto generated or not.

Currently, this page is for frequently used commands and is manually maintained at site2/docs/reference-cli-tools.md. 
As shown below, the comprehensive list is auto-generated at https://pulsar.apache.org/tools/.
image

**Comment by Shawyeok** [comment]
> > > Hi @Shawyeok, will you update the docs in a follow-up PR?
> > 
> > 
> > @momo-jun I'm not sure this doc [reference-cli-tools](https://pulsar.apache.org/docs/next/reference-cli-tools#pulsar-client) is auto generated or not.
> 
> Currently, this page is for frequently used commands and is manually maintained at site2/docs/reference-cli-tools.md. As shown below, the comprehensive list is auto-generated at https://pulsar.apache.org/tools/. image

Thanks for your detailed answer, now I've committed the doc update.

File: site2/docs/reference-cli-tools.md [hunk 1]
```diff
@@ -328,8 +328,8 @@ Options
 
 |Flag|Description|Default|
 |---|---|---|
-|`-f`, `--files`|Comma-separated file paths to send; either -m or -f must be specified|[]|
-|`-m`, `--messages`|Comma-separated string of messages to send; either -m or -f must be specified|[]|
+|`-f`, `--files`|Comma separated file paths to send, either -m or -f or --null must be specified.|[]|
```
Comment (momo-jun) [inline site2/docs/reference-cli-tools.md:331]: Comma-separated file paths to send. At least one of the following values must be specified: `-m`, `-f` or `--null` .

Comment (momo-jun) [inline site2/docs/reference-cli-tools.md:331]: Comma-separated file paths to send. At least one of the following values must be specified: `-m`, `-f` or `--null` .

File: site2/docs/reference-cli-tools.md [hunk 2]
```diff
@@ -328,8 +328,8 @@ Options
...
+|`-m`, `--messages`|Messages to send, either -m or -f or --null must be specified. The default separator is comma|[]|
```
Comment (momo-jun) [inline site2/docs/reference-cli-tools.md:332]:
```suggestion
|`-m`, `--messages`|Messages to send. At least one of the following values must be specified: `-m`, `-f` or `--null`. The default separator is a comma. |[]|
```
```suggestion
|`-m`, `--messages`|Messages to send, either -m or -f or --null must be specified. The default separator is comma|[]|
```

Comment (momo-jun) [inline site2/docs/reference-cli-tools.md:332]:
```suggestion
|`-m`, `--messages`|Messages to send. At least one of the following values must be specified: `-m`, `-f` or `--null`. The default separator is a comma. |[]|
```
```suggestion
|`-m`, `--messages`|Messages to send, either -m or -f or --null must be specified. The default separator is comma|[]|
```

File: site2/docs/reference-cli-tools.md (hunk 1 above)
Comment (momo-jun) [reply site2/docs/reference-cli-tools.md:331]:
```suggestion
|`-f`, `--files`|Comma-separated file paths to send. At least one of the following values must be specified: `-m`, `-f` or `--null`.|[]|
```

Comment (momo-jun) [reply site2/docs/reference-cli-tools.md:331]:
```suggestion
|`-f`, `--files`|Comma-separated file paths to send. At least one of the following values must be specified: `-m`, `-f` or `--null`.|[]|
```

File: site2/docs/reference-cli-tools.md [hunk 3]
```diff
@@ -339,6 +339,7 @@ Options
 |`-p`, `--properties`|Properties to add. If you want to add multiple properties, use the comma as the separator, e.g. `k1=v1,k2=v2`.| |
 |`-ekn`, `--encryption-key-name`|The public key name to encrypt payload.| |
 |`-ekv`, `--encryption-key-value`|The URI of public key to encrypt payload. For example, `file:///path/to/public.key` or `data:application/x-pem-file;base64,*****`.| |
+|`--null`|Send a message with the body is null|false|
```
Comment (momo-jun) [inline site2/docs/reference-cli-tools.md:342]:
```suggestion
|`--null`|Send a message with a null body.|false|
```

**Review by momo-jun — APPROVED** [review]
Doc changes look good to me.
//cc @Anonymitaet

**Review by Jason918 — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
Closed as stale and conflict. Feel free to rebase and resubmit the patch.

**Pull request closed without merging**